common/djangoapps/util/enterprise_helpers.py

```diff
@@ -203,6 +203,12 @@ def inner(request, course_id, *args, **kwargs):
         # Redirect to the consent URL, if consent is required.
         consent_url = get_enterprise_consent_url(request, course_id)
         if consent_url:
+            user = request.user
+            LOGGER.warning(
+                u'User %s cannot access the course %s because they have not granted consent',
+                user.get('real_user', user),
+                course_id,
+            )
             return redirect(consent_url)
```

Comment:
I looked at this some more, and the default is exactly as I described (it will redirect back to the original path). So I think this is all set.